### ISSUE TITLE
Autocompletion wants me to build the project

### ISSUE THREAD

**Issue by mrphil2105** [AvaloniaUI/AvaloniaVSCode#123]
### Describe the bug

When I try to use autocomplete in VSCode in a .axaml file the only option that shows up is "Build the project". Building the project does not help. I am actually using Neovim with the LSP, but I tried it in VSCode as well, and the issue persists.

### To Reproduce

The project consists of a root folder with an `.sln` file and 4 other folders containing `.csproj` files. One of the folders contains an Avalonia project. This is the project I am trying to use the extension with: [BachelorProject](https://github.com/mrphil2105/BachelorProject/)

### Avalonia for VS Code

v0.0.26

### Avalonia version

v11.0.10 and tested v11.0.2 as well

### VS Code version

v1.88.1

### Relevant log output

```shell
2024-04-17 17:36:20.659 [info] Creating language service
2024-04-17 17:36:20.659 [info] Starting Avalonia Language Server...
2024-04-17 17:36:20.659 [info] [Trace - 17:36:20] Sending request 'initialize - (0)'.
2024-04-17 17:36:21.459 [info] [Trace - 17:36:21] Received response 'initialize - (0)' in 808ms.
2024-04-17 17:36:21.459 [info] [Trace - 17:36:21] Sending notification 'initialized'.
2024-04-17 17:36:21.470 [info] [Trace - 17:36:21] Sending notification 'workspace/didChangeConfiguration'.
2024-04-17 17:36:21.519 [info] [Trace - 17:36:21] Received request 'client/registerCapability - (1)'.
2024-04-17 17:36:21.519 [info] [Trace - 17:36:21] Sending response 'client/registerCapability - (1)'. Processing request took 2ms
2024-04-17 17:36:21.519 [info] [Trace - 17:36:21] Sending notification 'textDocument/didOpen'.
2024-04-17 17:36:21.541 [info] [Trace - 17:36:21] Received request 'workspace/configuration - (2)'.
2024-04-17 17:36:21.541 [info] [Trace - 17:36:21] Sending response 'workspace/configuration - (2)'. Processing request took 1ms
2024-04-17 17:36:21.595 [info] [Trace - 17:36:21] Received request 'workspace/configuration - (3)'.
2024-04-17 17:36:21.596 [info] [Trace - 17:36:21] Sending response 'workspace/configuration - (3)'. Processing request took 0ms
2024-04-17 17:36:21.643 [info] [Trace - 17:36:21] Received request 'workspace/configuration - (4)'.
2024-04-17 17:36:21.643 [info] [Trace - 17:36:21] Sending response 'workspace/configuration - (4)'. Processing request took 0ms
2024-04-17 17:36:21.668 [info] [Trace - 17:36:21] Received notification 'window/logMessage'.
2024-04-17 17:36:21.668 [info] [Info  - 17:36:21] AvaloniaLanguageServer.Handlers.TextDocumentSyncHandler: ** DidOpenText: file:///home/mrphil2105/CSharp%20Projects/BachelorProject/AvaloniaApp/Views/MainView.axaml | Uri='file:///home/mrphil2105/CSharp%20Projects/BachelorProject/AvaloniaApp/Views/MainView.axaml'
2024-04-17 17:36:25.105 [info] [Trace - 17:36:25] Sending notification 'textDocument/didChange'.
2024-04-17 17:36:25.893 [info] [Trace - 17:36:25] Sending notification 'textDocument/didChange'.
2024-04-17 17:36:25.901 [info] [Trace - 17:36:25] Sending request 'textDocument/completion - (1)'.
2024-04-17 17:36:25.974 [info] [Trace - 17:36:25] Received response 'textDocument/completion - (1)' in 74ms.
2024-04-17 17:36:25.992 [info] [Trace - 17:36:25] Sending request 'completionItem/resolve - (2)'.
2024-04-17 17:36:26.019 [info] [Trace - 17:36:26] Received response 'completionItem/resolve - (2)' in 27ms.
2024-04-17 17:36:28.497 [info] [Trace - 17:36:28] Sending notification 'textDocument/didChange'.
2024-04-17 17:36:28.551 [info] [EXT - INFO] Solution data path path: /tmp/Apachi.sln.json
2024-04-17 17:36:28.594 [info] parser process args: /bin/sh,-c,dotnet "/home/mrphil2105/.vscode-oss/extensions/avaloniateam.vscode-avalonia-0.0.26-universal/solutionParserTool/SolutionParser.dll" "/home/mrphil2105/CSharp Projects/BachelorProject/Apachi.sln" --sdk 7.0
2024-04-17 17:36:28.915 [info] parser process exited 255
```

### Additional context

In Neovim I used the following to set up the LSP:
```lua
local configs = require("lspconfig.configs")
local root_pattern = require("lspconfig.util").root_pattern
if not configs.avalonia then
    configs.avalonia = {
        default_config = {
            cmd = { "dotnet", "/path/to/AvaloniaLanguageServer.dll" },
            root_dir = root_pattern("*.csproj"),
            filetypes = { "axaml", "xml" },
        },
    }
end

require("lspconfig").avalonia.setup(coq.lsp_ensure_capabilities {
    on_attach = on_attach, -- My attach function.
    settings = {},
})
```

P.S. In the log output I can see that it runs with `--sdk 7.0`. Does this mean I cannot use .NET 8.0 and have to use .NET 7.0?

**Comment by 10Peppermill** [comment]
I was having a similar issue and fixed it by following the trouble shooting in item two of this [disucssion](https://github.com/AvaloniaUI/Avalonia/discussions/12711#discussion-5570499). To save you a click; Close VSCode, navigate to project folder in terminal, call `dotnet build`, reopen in vscode.

**Comment by mrphil2105** [comment]
> I was having a similar issue and fixed it by following the trouble shooting in item two of this [disucssion](https://github.com/AvaloniaUI/Avalonia/discussions/12711#discussion-5570499). To save you a click; Close VSCode, navigate to project folder in terminal, call `dotnet build`, reopen in vscode.

I mentioned in my initial comment above that building the project does not help. So this suggestion does not help, unfortunately. I think the LSP fails to detect that the project has been built.

**Comment by lucadecamillis** [comment]
I had the same issue here. What helped me was to install the following two extensions in vscode:

- https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit
- https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.vscodeintellicode-csharp

After that autocomplete started to work.

**Comment by mrphil2105** [comment]
> I had the same issue here. What helped me was to install the following two extensions in vscode:
> 
>     * https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit
> 
>     * https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.vscodeintellicode-csharp
> 
> 
> After that autocomplete started to work.

I wonder what the equivalent fix is in Neovim, as I don't actually use VSCode for any development.

**Comment by DolmaAndKebab** [comment]
Tried to do the same thing...

```lua
vim.api.nvim_create_autocmd({"BufNewFile", "BufRead"}, {
	pattern={"*.axaml"},
	group=vim.api.nvim_create_augroup("axaml-ls", { clear = true } ),
	callback = function()
		local lsp_bin_dir = "~/.vscode/extensions/avaloniateam.vscode-avalonia-0.0.31/avaloniaServer/"
		local lsp_bin_path = "~/.vscode/extensions/avaloniateam.vscode-avalonia-0.0.31/avaloniaServer/AvaloniaLanguageServer.dll";
		if vim.fn.isdirectory(lsp_bin_dir) then
			vim.cmd.setfiletype("xml")
			vim.lsp.start({
				name = "axaml",
				cmd = {"dotnet", lsp_bin_path, "--stdio"},
				root_dir = vim.fn.getcwd(),
				capabilities = capabilities,
			})
		else
			print("Failed to start Avalona Language Server because the lsp bin dir is incorrect!")
		end
	end
})
``

And it doesnt work however I managed to fix it in vscode..

It seems that the LSP or maybe the extension I am unsure.. But it seems to be asking the .Net Install Extension in vscode to install dotnet 8 and probably something else who knows I fixed it in vscode by giving my dotnet bin path to .Net Install.

I pretty much opened up "User profile settings (JSON)"

**Comment by mrphil2105** [comment]
> Tried to do the same thing...
> 
> ```lua
> vim.api.nvim_create_autocmd({"BufNewFile", "BufRead"}, {
> 	pattern={"*.axaml"},
> 	group=vim.api.nvim_create_augroup("axaml-ls", { clear = true } ),
> 	callback = function()
> 		local lsp_bin_dir = "~/.vscode/extensions/avaloniateam.vscode-avalonia-0.0.31/avaloniaServer/"
> 		local lsp_bin_path = "~/.vscode/extensions/avaloniateam.vscode-avalonia-0.0.31/avaloniaServer/AvaloniaLanguageServer.dll";
> 		if vim.fn.isdirectory(lsp_bin_dir) then
> 			vim.cmd.setfiletype("xml")
> 			vim.lsp.start({
> 				name = "axaml",
> 				cmd = {"dotnet", lsp_bin_path, "--stdio"},
> 				root_dir = vim.fn.getcwd(),
> 				capabilities = capabilities,
> 			})
> 		else
> 			print("Failed to start Avalona Language Server because the lsp bin dir is incorrect!")
> 		end
> 	end
> })
> ```
> 
> And it doesnt work however I managed to fix it in vscode..
> 
> It seems that the LSP or maybe the extension I am unsure.. But it seems to be asking the .Net Install Extension in vscode to install dotnet 8 and probably something else who knows I fixed it in vscode by giving my dotnet bin path to .Net Install.
> 
> I pretty much opened up "User profile settings (JSON)"

I tried in VSCode and gave the .NET path. It still didn't work though. And even if it did it wouldn't help me since I never use VSCode.

**Comment by DolmaAndKebab** [comment]
Heres what I did to fix it.

# Step 1

Open up settings.json..

Ctrl + Shift + p
Then search for
Open User Settings (JSON)

# Step 2

Add the following to your settings.json

```json
 }
    "dotnetAcquisitionExtension.existingDotnetPath": [
        {"extensionId": "AvaloniaTeam.vscode-avalonia", "path": "/usr/bin/dotnet"}
    ],
    "dotnetAcquisitionExtension.installTimeoutValue": 180,
    "editor.minimap.enabled": false
}
```

**REPLACE /usr/bin/dotnet WITH THE BIN PATH TO DOTNET**

In order to find the dotnet bin path open up the terminal and run
whereis dotnet and get the path and replace /usr/bin/dotnet

**Comment by mrphil2105** [comment]
Now I wonder how you can replicate this in Neovim. If I didn't use Neovim I'd use Rider instead of VSCode, which has official Avalonia support.

**Comment by DolmaAndKebab** [comment]
I made it work once but then I did something by mistake and it broke..

I tried remaking it and even moving the LSP.

```js
vim.api.nvim_create_autocmd({"BufNewFile", "BufRead"}, {
	pattern={"*.axaml"},
	group=vim.api.nvim_create_augroup("axaml-ls", { clear = true } ),
	callback = function()
		if vim.fn.isdirectory("~/.config/nvim/LSP-Servers/") then
			local lsp_bin_dir = "~/.config/nvim/LSP-Servers/avaloniaServer"
			local lsp_bin_path = "~/.config/nvim/LSP-Servers/avaloniaServer/AvaloniaLanguageServer.dll";
			if vim.fn.isdirectory(lsp_bin_dir) then
				vim.cmd.setfiletype("xml")
				vim.lsp.start({
					name = "axaml",
					cmd = {"dotnet", lsp_bin_path},
					root_dir = vim.fn.getcwd(),
					capabilities = capabilities,
				})
			else
				print("Failed to start Avalona Language Server because the lsp bin dir does not exist!")
			end
		else
			print("Failed to start Avalonia Language Server. LSP-Servers directory does not exist!")
		end
	end
})
```

But sadly it just responds with something like "Client 1 exit with 1 and quit with 0"

**Comment by DolmaAndKebab** [comment]
Well  I guess some how someone made it work..

https://github.com/Johanw123/avalonia.nvim

**Comment by mrphil2105** [comment]
> Well I guess some how someone made it work..
> 
> https://github.com/Johanw123/avalonia.nvim

That one is only for previewer

**Comment by DolmaAndKebab** [comment]
It shows a example for LSP.

**Comment by DolmaAndKebab** [comment]
I found out that all you need to do is press enter when it tells you to build the project and it fixes it..

**Comment by DolmaAndKebab** [comment]
I messed around a bit and managed to get some of the LSP to work now the LSP just keeps recommending the same completion according to the LSP logs.

```lua
local Avalonia_LSP_Path_dir = vim.fn.expand("~/.config/nvim/LSP/avaloniaServer/")
local Avalonia_LSP_Path = vim.fn.expand("~/.config/nvim/LSP/avaloniaServer/AvaloniaLanguageServer.dll")

function _ISFILE(name, root)
	if type(name) ~= "string" then
		return false
	end
	if type(root) ~= "string" then
		return false
	end
	return vim.fn.findfile(name, root)
end

vim.api.nvim_create_autocmd({ "BufNewFile", "BufRead" }, {
	pattern = { "*.axaml" },
	group = vim.api.nvim_create_augroup("Avalonia_LSP", { clear = true }),
	callback = function()
		if vim.fn.isdirectory(Avalonia_LSP_Path_dir) then
			if not _ISFILE("AvaloniaLanguageServer.dll", Avalonia_LSP_Path_dir) then
				print("Avalonia LSP does not exist!")
				return
			end
			vim.cmd.setfiletype("xml")

			-- Setting the root_dir
			local root_dir = lspconfig.util.root_pattern("*.csproj", "*.sln")(vim.fn.getcwd())
			if not root_dir then
				print("Root directory not found!")
				return
			end

			-- For debugging
			print("Root directory: " .. root_dir)

			-- Building the project

			print("Building the project!")
			vim.cmd("!dotnet build")

			vim.lsp.start({
				name = "Avalonia_LS",
				cmd = { "dotnet", Avalonia_LSP_Path },
				root_dir = root_dir,
				filetypes = { "axaml", "xml" },
				capabilities = capabilities, -- Your capabilities
			})
			-- For debugging
			vim.lsp.set_log_level("debug")
			local log_path = vim.lsp.get_log_path()
			print("LSP log path: " .. log_path)
		else
			print("Avalonia LSP Dir not found!.. Provided path: " + Avalonia_LSP_Path_dir)
		end
	end,
})

vim.api.nvim_create_user_command("BuildProject", function()
	vim.cmd("!dotnet build")
end, {})

```

**Comment by MeKebabMan** [comment]
# Fix!

Open up vscode and when it asks you to Build the project just press enter and let it do it's thing after that close vscode and open up neovim with your avalonia LSP config and it should work!

**Comment by MeKebabMan** [comment]
I found out that the Avalonia LSP requires some files in the Bin and OBJ directories. So its able to create those files in VScode but it is not able to in Neovim so what you can do is create a template that already contains the files required for the LSP and each time you use Avalonia instead of doing dotnet new avalonia.app you can just do git clone https://user/my-template. The template should contain the right files for the LSP to work correctly! This can be easily fixed in neovim now let me try explaining how it should work in neovim!

So what you can do is every time the Avalonia LSP Attaches and there's a axaml file you can use LUA to create the required files for the avalonia LSP meaning the user or you will never have to tab into vscode just to build the project!

# How I found this out??

I started by creating a avalonia app and then tracking files being created & deleted I found out that some files were modified and new files were created after the vscode extension attempted to build the project..

I managed to get it working in neovim!!!!

**Comment by mrphil2105** [comment]
Selecting 'Build the project' in VSCode and pressing Enter did not work for me. But then I noticed that it runs the build command with `--sdk 7.0` and I only had 8.0 installed. Installing 7.0 made the build command work (keep in mind that a regular build with `dotnet build` did not work as mentioned before). This made it create the files in obj you mentioned and now the LSP works in Neovim! A bit strange since the VSCode extension page mentions that you need 8.0 and not 7.0. Any idea why it does not use 8.0 and whether building my project with 8.0 will break the LSP somehow?

**Comment by MeKebabMan** [comment]
I mean you can just create a bash script to create the files that it requires.. But who in the world has enough time to figure all of that out? its been 3 weeks and I haven't tested it and to be honest if it's this annoying to get a LSP to work then I might aswell just use GTK4.

**Comment by VNNCC** [comment]
Just came across this issue.

**I am using VSCode and .NET 8.0.400-preview.0.24324.5 and believe that my _fix_ will only be applicable for those using a preview version.**

### TL;DR

Identified and fixed issues with the `Build project` button, preview feature, and XAML IntelliSense. The fix involved updating the `Microsoft.Build.dll` libraries in the solution parser tool directory of the extension (`...\.vscode-insiders\extensions\avaloniateam.vscode-avalonia-0.0.31\solutionParserTool\`), which resolved the errors.

### Detailed Description

#### Issues Addressed by the Fix

- `Build project` button not functioning
- Preview feature not working
- XAML IntelliSense issues

#### Fix

Upon reviewing the logs, I identified the following entries:

```
[info] Command avalonia.previewProcess, [SomeFile].axaml
[info] [EXT - INFO] Solution data path path: ../[SolutionName].sln.json
[info] parser process args: C:\Windows\system32\cmd.exe,/d,/s,/c,"dotnet "....vscode-insiders\extensions\avaloniateam.vscode-avalonia-0.0.31\solutionParserTool\SolutionParser.dll" "...[SolutionName].sln""
```

The entry for `Solution data path path` caught my attention. Upon inspecting it, I found the following content:

```json
{
  "solution": "../[SolutionName].sln",
  "projects": [],
  "files": []
}
```

This content was clearly incorrect.

To address this, I executed the following command:
```cmd
dotnet "...\.vscode-insiders\extensions\avaloniateam.vscode-avalonia-0.0.31\solutionParserTool\SolutionParser.dll" "...\[SolutionName].sln"
```

The result was an error:

```cmd
Error parsing project [ProjectName]: Invalid static method invocation syntax: "[MSBuild]::SubstringByAsciiChars($(MSBuildProjectFile), 0, 8)". Method '[MSBuild]::SubstringByAsciiChars' not found. Static method invocation should be of the form: $([FullTypeName]::Method()), e.g. $([System.IO.Path]::Combine(`a`, `b`)). Check that all parameters are defined, are of the correct type, and are specified in the right order.  ...\dotnet\sdk\8.0.400-preview.0.24324.5\Microsoft.Common.CurrentVersion.targets
```

Upon examining the `Microsoft.Common.CurrentVersion.targets` file, I found the following:

```xml
<MSBuildCopyMarkerName Condition="'$(MSBuildCopyMarkerName.Length)' &gt; '17'">$([MSBuild]::SubstringByAsciiChars($(MSBuildProjectFile), 0, 8)).$([MSBuild]::StableStringHash($(MSBuildProjectFile)).ToString("X8"))</MSBuildCopyMarkerName>
```

I compared this with the version `.NET 8.0.206`:

```diff
-<MSBuildCopyMarkerName Condition="'$(MSBuildCopyMarkerName.Length)' &gt; '17'">$(MSBuildProjectFile.Substring(0,8)).$([MSBuild]::StableStringHash($(MSBuildProjectFile)).ToString("X8"))</MSBuildCopyMarkerName>
+<MSBuildCopyMarkerName Condition="'$(MSBuildCopyMarkerName.Length)' &gt; '17'">$([MSBuild]::SubstringByAsciiChars($(MSBuildProjectFile), 0, 8)).$([MSBuild]::StableStringHash($(MSBuildProjectFile)).ToString("X8"))</MSBuildCopyMarkerName>
```
As seen, there was a change.

I then examined the code to determine which library was used for project information collection, which was `Microsoft.Build.dll`.

By copying the .NET preview `Microsoft.Build.*.dll` libraries from `..\dotnet\sdk\8.0.400-preview.0.24324.5\` into the following directory `...\.vscode-insiders\extensions\avaloniateam.vscode-avalonia-0.0.31\solutionParserTool\` the `[SolutionName].sln.json` file was successfully generated, resolving the IntelliSense and preview issues.

**Comment by auxym** [comment]
Same experience here. Unfortunately, this means this extension is completely broken for me. No auto-complete and no preview.

Windows 11, dotnet SDK 8.0

I saw the following in the Avalonia client logs:

```
2024-11-19 16:07:55.015 [info] parser process args: C:\windows\system32\cmd.exe,/d,/s,/c,"dotnet "c:\Users\thef2102\.vscode\extensions\avaloniateam.vscode-avalonia-0.0.31\solutionParserTool\SolutionParser.dll" "c:\Users\thef2102\source\mrfsplus_code\SenSed\sensed-desktop\sensed-desktop.sln""
```

There's nothing more in the logs, but when I manually run this command from cmd, I get the following output:

```
Error parsing project sensed-desktop: Invalid static method invocation syntax: "[MSBuild]::SubstringByAsciiChars($(MSBuildProjectFile), 0, 8)". Method '[MSBuild]::SubstringByAsciiChars' not found. Static method invocation should be of the form: $([FullTypeName]::Method()), e.g. $([System.IO.Path]::Combine(`a`, `b`)). Check that all parameters are defined, are of the correct type, and are specified in the right order.  C:\Users\thef2102\AppData\Local\Microsoft\dotnet\sdk\8.0.404\Microsoft.Common.CurrentVersion.targets
{
  "solution": "c:\\Users\\thef2102\\source\\mrfsplus_code\\SenSed\\sensed-desktop\\sensed-desktop.sln",
  "projects": [],
  "files": []
}
```

**Comment by auxym** [comment]
Hi @MikeCodesDotNET I see the issue should be fixed, however the new version is not yet available from the VS Marketplace I believe? Could you please release a new version, so we can get the fix?

**Comment by auxym** [comment]
This issue appears to still exist and should be reopened please.

I managed to build and install the extension (after applying the fix in #140), and I still get the same behavior. Neither autocomplete or preview works. When trying to autocomplete, I get the "Build project" pop-up. When I click it, a few lines appear in the log (the solution parser appears to get invoked), but autocomplete still doesn't work after. I try to autocomplete again, and I just get the same "Build project" pop-up again.

Here's a log excerpt of what happens when I type `<` in a axaml file, get the "Build project pop-up" and click it:

```text
2024-11-20 16:19:48.099 [info] [Trace - 4:19:48 PM] Sending notification 'textDocument/didChange'.
2024-11-20 16:19:49.134 [info] [Trace - 4:19:49 PM] Sending notification 'textDocument/didSave'.
2024-11-20 16:19:49.139 [info] [Trace - 4:19:49 PM] Received notification 'window/logMessage'.
2024-11-20 16:19:49.139 [info] [Info  - 4:19:49 PM] AvaloniaLanguageServer.Handlers.TextDocumentSyncHandler: DidSaveTextDocumentParams OmniSharp.Extensions.LanguageServer.Protocol.Models.DidSaveTextDocumentParams | Request='OmniSharp.Extensions.LanguageServer.Protocol.Models.DidSaveTextDocumentParams'
2024-11-20 16:19:49.438 [info] [Trace - 4:19:49 PM] Sending notification 'workspace/didChangeWatchedFiles'.
2024-11-20 16:19:56.880 [info] [Trace - 4:19:56 PM] Sending notification 'textDocument/didChange'.
2024-11-20 16:19:57.860 [info] [Trace - 4:19:57 PM] Sending notification 'textDocument/didChange'.
2024-11-20 16:19:57.864 [info] [Trace - 4:19:57 PM] Sending request 'textDocument/completion - (17)'.
2024-11-20 16:19:57.918 [info] [Trace - 4:19:57 PM] Received response 'textDocument/completion - (17)' in 54ms.
2024-11-20 16:19:57.932 [info] [Trace - 4:19:57 PM] Sending request 'completionItem/resolve - (18)'.
2024-11-20 16:19:57.934 [info] [Trace - 4:19:57 PM] Received response 'completionItem/resolve - (18)' in 2ms.
2024-11-20 16:19:58.905 [info] [Trace - 4:19:58 PM] Sending notification 'textDocument/didSave'.
2024-11-20 16:19:58.918 [info] [Trace - 4:19:58 PM] Received notification 'window/logMessage'.
2024-11-20 16:19:58.918 [info] [Info  - 4:19:58 PM] AvaloniaLanguageServer.Handlers.TextDocumentSyncHandler: DidSaveTextDocumentParams OmniSharp.Extensions.LanguageServer.Protocol.Models.DidSaveTextDocumentParams | Request='OmniSharp.Extensions.LanguageServer.Protocol.Models.DidSaveTextDocumentParams'
2024-11-20 16:19:59.211 [info] [Trace - 4:19:59 PM] Sending notification 'workspace/didChangeWatchedFiles'.
2024-11-20 16:20:03.968 [info] [Trace - 4:20:03 PM] Sending notification 'textDocument/didChange'.
2024-11-20 16:20:04.112 [info] [EXT - INFO] Solution data path path: C:\Users\thef2102\AppData\Local\Temp\sensed-desktop.sln.json
2024-11-20 16:20:04.197 [info] parser process args: C:\windows\system32\cmd.exe,/d,/s,/c,"dotnet "c:\Users\thef2102\.vscode\extensions\avaloniateam.vscode-avalonia-0.0.31\solutionParserTool\SolutionParser.dll" "c:\Users\thef2102\source\mrfsplus_code\SenSed\sensed-desktop\sensed-desktop.sln""
2024-11-20 16:20:05.025 [info] [Trace - 4:20:04 PM] Sending notification 'textDocument/didSave'.
2024-11-20 16:20:05.039 [info] [Trace - 4:20:04 PM] Received notification 'window/logMessage'.
2024-11-20 16:20:05.039 [info] [Info  - 4:20:04 PM] AvaloniaLanguageServer.Handlers.TextDocumentSyncHandler: DidSaveTextDocumentParams OmniSharp.Extensions.LanguageServer.Protocol.Models.DidSaveTextDocumentParams | Request='OmniSharp.Extensions.LanguageServer.Protocol.Models.DidSaveTextDocumentParams'
2024-11-20 16:20:05.070 [info] parser process exited 0
2024-11-20 16:20:05.342 [info] [Trace - 4:20:05 PM] Sending notification 'workspace/didChangeWatchedFiles'.
```

Note: If I manually invoke the solutionparser as above, I no longer get the MSBuild method error. What happens is just that the program hangs and never completes. I'm not sure if this is expected behavior.

**Comment by TheOldMiguel** [comment]
I tried everything mentioned in this thread to get autocomplete and the preview to work. But it just says "Build the project" each time. I even tried building the most recent SolutionParser myself (there was apparently a fix done last week) and copying it into the solution parser path of the VsCode Extension. Unfortunately, none of this worked. I'm very interested in Avalonia and I hope there will be a new preview-version of the extension soon.

**Comment by auxym** [comment]
> I tried everything mentioned in this thread to get autocomplete and the preview to work. But it just says "Build the project" each time. I even tried building the most recent SolutionParser myself (there was apparently a fix done last week) and copying it into the solution parser path of the VsCode Extension. Unfortunately, none of this worked. I'm very interested in Avalonia and I hope there will be a new preview-version of the extension soon.

Have you tried building and installing the extension from the master branch? It works for me since #135 and #140 are merged. But yes I agree, a new version should really be released.

**Comment by MikeCodesDotNET** [comment]
I've just pushed an update to the marketplace.
